packages/app/src/app/pages/Sandbox/Editor/Workspace/Project/Keywords.tsx

```diff
@@ -1,77 +1,74 @@
-import React from 'react';
-import { inject, hooksObserver, clone } from 'app/componentConnectors';
+import React, { FunctionComponent } from 'react';
 import Tags from '@codesandbox/common/lib/components/Tags';
 import getTemplateDefinition from '@codesandbox/common/lib/templates';
+
+import { clone } from 'app/componentConnectors';
 import { EditableTags } from 'app/components/EditableTags';
+import { useOvermind } from 'app/overmind';
+
 import { Item } from './elements';
 
-interface IKeywordsProps {
+type Props = {
   editable?: boolean;
-  store: any;
-  signals: any;
-}
-
-export const Keywords = inject('store', 'signals')(
-  hooksObserver(
-    ({
-      editable,
-      store: {
-        editor: {
-          currentSandbox: { template, tags },
-        },
-        workspace: {
-          tags: { tagName },
-        },
+};
+export const Keywords: FunctionComponent<Props> = ({ editable }) => {
+  const {
+    actions: {
+      notificationAdded,
+      workspace: { tagAdded, tagChanged, tagRemoved },
+    },
+    state: {
+      editor: {
+        currentSandbox: { template, tags },
       },
-      signals: {
-        notificationAdded,
-        workspace: { tagAdded, tagChanged, tagRemoved },
+      workspace: {
+        tags: { tagName },
       },
-    }: IKeywordsProps) => {
-      const changeTags = (newTags: string[], removedTags: string[]) => {
-        if (tags.length > 5) {
-          notificationAdded('You can have a maximum of 5 tags', 'error');
-          return;
-        }
+    },
+  } = useOvermind();
 
-        const tagWasRemoved =
-          newTags.length < tags.length && removedTags.length === 1;
+  if (tags.length === 0 && !editable) {
+    return null;
+  }
 
-        if (tagWasRemoved) {
-          removedTags.forEach(tag => {
-            tagRemoved({ tag });
-          });
-        } else {
-          tagAdded();
-        }
-      };
+  const changeTags = (newTags: string[], removedTags: string[]) => {
+    if (tags.length > 5) {
+      notificationAdded('You can have a maximum of 5 tags', 'error');
```

Comment:
@christianalfoni `notificationAdded` has the following signature:
https://github.com/codesandbox/codesandbox-client/blob/cb965d440eb569d569e97ded524d54d142f241ad/packages/app/src/app/overmind/actions.ts#L28-L38

Should I still use `notificationAdded` or is another method preferred? 🤔

Reply:
That is the way, I think time alive was supposed to be an optional number because its all over the code without it

@christianalfoni

Reply:
Then `timeAlive` should have a default of `1` too I guess? 🤔

Reply:
Ah, yeah, but in this case we have logic inside the component controllin how many tags can be added. This should be moved to `tagAdded` action. That should check number of tags and then directly use the `notifcationsToast.error` method.

This just ensures that if we ever change this component or allow adding tags from somewhere else we have this logic centralized :)

Reply:
Actually... just create a new action `tagsChanged` and use that directly. Move the logic into that action. Remove `tagRemoved` and `tagAdded` 😄